tfhe/src/c_api/shortint/server_key/scalar_add.rs

```diff
@@ -6,17 +6,17 @@ use super::{ShortintCiphertext, ShortintServerKey};
 #[no_mangle]
 pub unsafe extern "C" fn shortint_server_key_smart_scalar_add(
     server_key: *const ShortintServerKey,
-    ct_left: *mut ShortintCiphertext,
+    ct_left: *const ShortintCiphertext,
```

Comment:
this should still be mut right ? as discussed for the smart stuff

code below should probably be updated as well

Reply:
the first version you describe sounds like a default operation, i.e. output is always clean
the second is a smart operation as far as I can tell

Reply:
I see what you mean
But it seems to me that the first version can give better performance than the second one if the input is never reused
So with this definition of smart operation, using smart operations can be more costly than using default ones (which is not intuitive)

I think the second version is more consistent with the rest of the codebase so I'll use it

Reply:
indeed, though smart ops have this approach of cleaning inputs to hopefully gain time on later ops if re-used

that's why default ops have been made default, easier to reason about and not that bad performance characteristics ! 🙂